scipy/interpolate/_bsplines.py

```diff
@@ -314,6 +322,14 @@ def __call__(self, x, nu=0, extrapolate=None):
         x = np.asarray(x)
         x_shape, x_ndim = x.shape, x.ndim
         x = np.ascontiguousarray(x.ravel(), dtype=np.float_)
+
+        # With periodic extrapolation we map x to the segment
+        # [self.t[k], self.t[-k]].
+        if extrapolate == 'periodic':
+            x = self.t[self.k] + (x - self.t[self.k]) % (self.t[-(self.k)] -
```

Comment:
It did got confusing because of my error. However, it seems right now. I will make the corrections and all the other changes in a day or two.